fix: only compare minor or patch number if parent is equal to latest

cpradog opened the pull request:
In this scenario:

- installed version: 5.0.0
- latest "stable" version: 4.3.2

minor and patch numbers should not be highlighted as if there is an update available because major version is higher.

cpradog committed:
fix: package version over public release version

Only compare minor or patch version numbers if parent number is equal.

meain commented on rplugin/node/vim-package-info/diff.js:
```diff
@@ -35,27 +35,36 @@ function colorizeDiff(current, latest, hl) {
-      [".", hl],
-      [l.patch, "VimPackageInfoPatch"],
-    ];
+  } else if (c !== null && parseInt(l.major === parseInt(c.major))) {
```
The brackets seem to be misaligned.
```suggestion
  } else if (c !== null && parseInt(l.major) === parseInt(c.major)) {
```

cpradog committed:
Update rplugin/node/vim-package-info/diff.js

Co-authored-by: Abin Simon <abinsimon10@gmail.com>

cpradog committed:
major refactor of version diff algorithm.

Now semver-utils is no longer required and IMHO is easier to follow and
reason about.

Changes:

- removed hacks and workarounds for semver-utils
- version release and build parts are now shown in final highlight
- separate diff calculation from highlight generation

meain commented:
Thanks a lot for the refactor.
